### PR TITLE
Feat/crashlytics

### PR DESCRIPTION
**Pull request by simon-the-shark**
so this adds firebase services:
- crashlytics
- performance monitor
- basic analytics (I think, without any custom events)
- push notifications (basic setup, we can only send notifications from console)

When it comes to testing, android seems fine.

on IOS I'm testing on emulator now (I don't have device with me) - but basically push notifications and deep links still don't work. We'll se on real iphone maybe

### CODE REVIEW

File: lib/features/guide_view/guide_view.dart [hunk 1]
```diff
@@ -19,11 +22,15 @@ import "widgets/guide_grid.dart";
...
 
 @RoutePage()
-class GuideView extends StatelessWidget {
+class GuideView extends HookWidget {
   const GuideView({super.key});
 
   @override
   Widget build(BuildContext context) {
+    useEffectOnInit(() {
+      Future.microtask(requestFCMPermission);
```
Comment (mikolaj-jalocha) [inline lib/features/guide_view/guide_view.dart:31]: What's the purpose of this code here? (why here)

**Review by mikolaj-jalocha — APPROVED** [review]
LGTM! 

I hope iOS'd be fine. 

Left one questions of mine.

File: lib/features/guide_view/guide_view.dart (hunk 1 above)
Comment (simon-the-shark) [reply lib/features/guide_view/guide_view.dart:31]: Yeah, I know this seems funny, but I just didn't want to show it as first things you enter the app and thought that once you enter the guide tab seems a good moment to ask for permissions. Just random idea. But we can change that if you have a better idea where to place it

Comment (simon-the-shark) [reply lib/features/guide_view/guide_view.dart:31]: Yeah, I know this seems funny, but I just didn't want to show it as first things you enter the app and thought that once you enter the guide tab seems a good moment to ask for permissions. Just random idea. But we can change that if you have a better idea where to place it

**Comment by mikolaj-jalocha** [comment]
I like the idea. Maybe later we can configure some local notification or
sth like that.  Maybe on home screen some  banner saying 'please grant
permission etc' with nice look and don't disturbing. Sth like cookies on
non commercial sites :) But guide page imo looks good for this, at least
for now.

**Comment by simon-the-shark** [comment]
Yeah good feature idea for some future

**Pull request merged**